regtests/bin/matrix_divider_p.sh

```diff
@@ -0,0 +1,118 @@
+#       National Oceanic and Atmospheric Administration.  All rights          #
+#       reserved.  WAVEWATCH III is a trademark of the NWS.                   #
+#       No unauthorized use without permission.                               #
+#                                                                             #
+# --------------------------------------------------------------------------- #
+# --------------------------------------------------------------------------- #
+# 1.  clean up and definitions                                                #
+# --------------------------------------------------------------------------- #
+rm -r ../model?
+rm -r ../model??
+cp matrix matrix.tmp
+
+maxlist1=47
+maxlist2=91
+
+#Put the job requirement/spec in "before"
+sed -e "/run_test/,\$d" matrix.tmp > before
+#Put the list of tests in "list"
+command egrep 'ww3_tp2.14|ww3_tp2.17' matrix.tmp | cat >> list_heavy
+awk '!/ww3_tp2.14/' matrix.tmp > tmpfile && mv tmpfile matrix.tmp
+awk '!/ww3_tp2.17/' matrix.tmp > tmpfile && mv tmpfile matrix.tmp
+command egrep 'mpirun|mpiexec|MPI_LAUNCH' matrix.tmp | cat >> list_mpi
+awk '!/mpirun|mpiexec|MPI_LAUNCH/' matrix.tmp > tmpfile && mv tmpfile matrix.tmp
+split -dl $maxlist1 list_mpi list_mpi_
+rm list_mpi
+matrixno1=$(ls list_mpi_* | wc -l)
+echo "Total nummber of matrix with parallel tests = $(($matrixno1 + 1)); each includes $maxlist1 tests"
+command egrep 'run_test' matrix.tmp | cat >> list_serial
+split -dl $maxlist2 list_serial list_serial_
+rm list_serial
+matrixno2=$(ls list_serial_* | wc -l)
+echo "Total nummber of matrix with serial test = $matrixno2; each includes $maxlist2 tests"
+rm matrix.tmp
+
+# --------------------------------------------------------------------------- #
+# 2.  Divide and dump in subsets                                              #
+# --------------------------------------------------------------------------- #
+# parallel jobs
+count=0
+  for i in `seq -f '%02g' 0 "$((matrixno1 - 1))"`; do
+#echo $i
+#Replace matrix.out > matrix?.out, model > model?
+  (( count = count + 1 ))
+  cat before >> matrix$count
+  cat list_mpi_$i >> matrix$count
+  cp -r ../model ../model$count
+  sed -i 's/'matrix.out'/'matrix${count}.out'/gI' matrix$count
+  sed -i 's/'model'/'model${count}'/gI' matrix$count
+  echo "  echo ' '"                                                                     >> matrix$count
+  echo "  echo '     **************************************************************'"   >> matrix$count
+  echo "  echo '     *  end of WAVEWATCH III matrix$count of regression tests     *'"   >> matrix$count
+  echo "  echo '     **************************************************************'"   >> matrix$count
+  echo "  echo ' '"                                                                     >> matrix$count
+  echo "rm -r ../model$count"                                                              >> matrix$count
```

Comment:
Instead of removing/cleaning the extra model directories in the top level script, which usually comes up as an error and since we remove the extra model directory in this script, maybe in the matrix? script, you check if the model? dir exists, if it does, remove and copy a new model? from model and then remove at the end of the script.    That way if for whatever reason you have to re-run the matrix? You don't have to re-run the top level script?

Reply:
@JessicaMeixner-NOAA Done, thanks for the suggestion. I went one step further and also put an if statement to remove matrix? if any.

Reply:
@JessicaMeixner-NOAA could you refer to the line you still like a change?